File: First Unique Character in a String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        bitset<26>p,q;
        for(int i=0;i<s.length();i++)
        {
            if(p[s[i]-'a']==0)
            {
                p[s[i]-'a']=1;
            }
            else if(p[s[i]-'a']==1)
            {
                q[s[i]-'a']=1;
            }
        }
        for(int i=0;i<s.length();i++)
        {
            if(q[s[i]-'a']==0)
            {
                return i;
            }
            
        }
        return -1;
    }
};

File: Golang Solutions/First Unique Character in a String.go
func firstUniqChar(s string) int {
    var p[26][2] bool
    for _,c := range s{
        if p[int(c)-int('a')][0]==false{
            p[int(c)-int('a')][0]=true
        }else if(p[int(c)-int('a')][0]==true){
            p[int(c)-int('a')][1]=true
        }
    }
    for i,c := range s{
        if p[int(c)-int('a')][1]==false{
            return i
        }
    }
    
    return -1
    
}